src/entities/createPool/createPoolV3/index.ts

```diff
@@ -0,0 +1,29 @@
+import { PoolType } from '@/types';
+import {
+    CreatePoolBase,
+    CreatePoolBuildCallOutput,
+    CreatePoolInput,
+} from '../types';
+import { CreatePoolWeightedV3 } from './weighted/createPoolWeighted';
+
+export class CreatePoolV3 implements CreatePoolBase {
+    // TODO: should we allow for custom create pool types?
```

Comment:
I'm not sure if this is important but might be worth bringing this up with Daniel?

Reply:
I think it's impossible to do so 😕 Because the CreatePoolInput would also be custom depending on the pool type, we cannot typify the buildCall input for custom Create Pool Types, the AddLiquidity and RemoveLiquidity can have custom types because the input type of the function doesn't change.

Reply:
Yeap. I added this comment a while ago to investigate, but we already discussed and reached a conclusion that this is not possible. You can remove the comment now.

Reply:
I was thinking about it today, and maybe we can handle this by using generic types if we need.
I'll ask Daniel if this is an interesting feature anyway.